clients/src/main/java/org/apache/kafka/clients/consumer/internals/ConsumerCoordinator.java

```diff
@@ -351,6 +359,10 @@ private Exception invokePartitionsRevoked(final SortedSet<TopicPartition> revoke
             final long startMs = time.milliseconds();
             listener.onPartitionsRevoked(revokedPartitions);
             sensors.revokeCallbackSensor.record(time.milliseconds() - startMs);
+            // remove the offset metadata cache for revoked partitions
+            for (TopicPartition revokedPartition: revokedPartitions) {
+                this.committedTopicPartitionOffsetsCache.remove(revokedPartition);
+            }
```

Comment:
The revoke and lost partition cases should also be tested, to verify we indeed remove them from cache.

Reply:
You are correct, we are missing this verification. I'll add them!

Reply:
done